test/test_config/test_configgenerator.cc

```diff
@@ -0,0 +1,226 @@
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+#include <config_manager.h>
+#include <config/config_generator.h>
+#include <fstream>
+#include <regex>
+
+using namespace ::testing;
+
+class ConfigGeneratorTest : public ::testing::Test {
+
+ public:
+  ConfigGeneratorTest() {};
+  virtual ~ConfigGeneratorTest() {};
+
+  virtual void SetUp() {
+    subject = new ConfigGenerator();
+    homePath = "/tmp";
+    configDir = ".config/gerbera";
+    prefixDir = "/usr/local/share/gerbera";
+    magicFile = "magic.file";
+  }
+
+  virtual void TearDown() {
+    delete subject;
+  };
+
+  std::string mockConfigXml(std::string mockFile) {
+    std::ifstream t(mockFile);
+    std::string str((std::istreambuf_iterator<char>(t)), std::istreambuf_iterator<char>());
+    return str;
+  }
+
+  ConfigGenerator *subject;
+  std::string homePath;
+  std::string configDir;
+  std::string prefixDir;
+  std::string magicFile;
+};
+
+#if defined(HAVE_FFMPEG) && defined(HAVE_FFMPEGTHUMBNAILER) && defined(HAVE_MYSQL) && defined(HAVE_MAGIC) && defined(HAVE_JS) && defined(ONLINE_SERVICES)
+TEST_F(ConfigGeneratorTest, GeneratesFullConfigXmlWithAllDefinitions) {
+  std::string mockXml = mockConfigXml("fixtures/mock-config-all.xml");
+
+  std::string result = subject->generate(homePath, configDir, prefixDir, magicFile);
+
+  // remove UUID, for simple compare...TODO: mock UUID?
+  std::regex reg("<udn>uuid:[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{12}</udn>");
```

Comment:
Now that `udn` is generated, it is difficult to test.  I simply removed the value and assert `<udn/>`

The **udn** is tested independently.